File: code_snippets/python-sdk/workflows/text_to_audio.py
from clarifai.client.workflow import Workflow


workflow_url = "https://clarifai.com/clarifai/Text/workflows/text-to-audio"
# Creating thr workflow
text_to_audio_workflow = Workflow(
    url=workflow_url,
    pat="YOUR_PAT",
)

# Getting the predictions
result = text_to_audio_workflow.predict_by_bytes(
    b"I love your product very much", input_type="text"
)

# Save the audio file
with open("output_audio.wav", mode="bx") as f:
    f.write(result.results[0].outputs[0].data.audio.base64)

File: code_snippets/python-sdk/workflows/textToAudio.ts
import { Workflow } from "clarifai-nodejs";
import fs from "fs";


const workflowUrl =
  "https://clarifai.com/clarifai/Text/workflows/text-to-audio";

// Creating the workflow
const textToAudioWorkflow = new Workflow({
  url: workflowUrl,
  authConfig: {
    pat: process.env.CLARIFAI_PAT,
  },
});

// Getting the predictions
const prediction = await textToAudioWorkflow.predictByBytes(
  Buffer.from("I love your product very much"),
  "text",
);

const imgB =
  prediction.resultsList?.[0]?.outputsList?.[0]?.data?.audio?.base64 ?? "";

// Save the audio file
fs.writeFileSync("audio.wav", imgB, "base64");
